From a74fc294eb7b2a766eb554024220617018450527 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:33 -0400
Subject: [tools] perf/completion: Strip dependency on _filedir

Message-id: <1408462094-14194-105-git-send-email-jolsa@redhat.com>
Patchwork-id: 88106
O-Subject: [PATCH RHEL7.1 BZ1131394 104/325] perf completion: Strip dependency on _filedir
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 7b6c48e16e5d312b0ebae78acfb3ff4f9c8c083c
Author: Ramkumar Ramachandra <artagnon@gmail.com>
Date: Thu Jul 4 18:11:27 2013 +0530

description
===========
_filedir is defined in the bash-completion package, but there is no need
to depend on it.  Instead, call complete with multiple -o arguments
before the -F argument like in git.git's completion script.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/bash_completion b/tools/perf/bash_completion
index b0cdd12..d2598be 100644
--- a/tools/perf/bash_completion
+++ b/tools/perf/bash_completion
@@ -54,9 +54,8 @@ _perf()
   subcmd=${COMP_WORDS[1]}
   opts=$($cmd $subcmd --list-opts)
   COMPREPLY=( $( compgen -W '$opts' -- "$cur" ) )
- # Fall down to list regular files
- else
-  _filedir
  fi
 } &&
-complete -F _perf perf
+
+complete -o bashdefault -o default -o nospace -F _perf perf 2>/dev/null \
+ || complete -o default -o nospace -F _perf perf
-- 
1.7.1